File: examples/docs-code-snippets/src/main/scala/examples/scala/recipes/WebShopRecipe.scala
package examples.scala.recipes

import com.ing.baker.recipe.scaladsl.{Event, Recipe}
import examples.scala.events.OrderPlaced
import examples.scala.interactions.{CancelOrder, CheckStock, ShipOrder}

object WebShopRecipe {
  val recipe: Recipe = Recipe("web-shop recipe")
    .withSensoryEvent(
      Event[OrderPlaced]
    )
    .withInteractions(
      CheckStock.interaction,
      ShipOrder.interaction
        .withRequiredEvent(
          Event[CheckStock.SufficientStock]
        ),
      CancelOrder.interaction
        .withRequiredEvent(
          Event[CheckStock.OrderHasUnavailableItems]
        )
    )
}

File: examples/docs-code-snippets/src/main/java/examples/java/recipes/WebShopRecipe.java
package examples.java.recipes;

import com.ing.baker.recipe.javadsl.InteractionDescriptor;
import com.ing.baker.recipe.javadsl.Recipe;
import examples.java.events.OrderPlaced;
import examples.java.interactions.CancelOrder;
import examples.java.interactions.CheckStock;
import examples.java.interactions.ShipOrder;

public class WebShopRecipe {

    public final static Recipe recipe = new Recipe("web-shop recipe")
        .withSensoryEvent(OrderPlaced.class)
        .withInteractions(
            InteractionDescriptor.of(CheckStock.class),
            InteractionDescriptor.of(ShipOrder.class)
                .withRequiredEvent(CheckStock.SufficientStock.class),
            InteractionDescriptor.of(CancelOrder.class)
                .withRequiredEvent(CheckStock.OrderHasUnavailableItems.class)
        );
}
